source/whats-new.txt

```diff
@@ -0,0 +1,176 @@
+     - Drops support for platform A, compiler B, or architecture C.
+
+.. _upcoming-breaking-changes:
+
+.. _version-3.10:
+
+What's New in 3.10
+------------------
+
+.. warning:: Build System Breaking Changes
+
+   The v3.10 driver introduces the following breaking changes to the build system:
+   
+   - Drops support for calling ``find_package(libbsoncxx)`` and ``find_package(libmongocxx)``.
+     Use ``find_package(bsoncxx)`` and ``find_package(mongocxx)``
+     instead.
+   - Drops support for the ``LIBBSONCXX_*`` and ``LIBMONGOCXX_*`` CMake variables
+     provided by the legacy CMake package config files. Use the ``mongo::bsoncxx_*`` and
+     ``mongo::mongocxx_*`` CMake targets instead.
+   - Removes the experimental C++ standard library, ``BSONCXX_POLY_USE_STD_EXPERIMENTAL``,
+     as a polyfill option.
+
+The v3.10.0 driver release includes the following new features:
+
+- Adds forward headers that provide non-defining declarations of ``bsoncxx`` and
+  ``mongocxx`` class types. The driver does not support user-defined forward declarations
+  of any library entity. To obtain the declaration or definition of a library entity,
+  always include the corresponding header.
+- The CMake option ``ENABLE_BSONCXX_USE_POLY_IMPLS`` allows selecting
+  ``bsoncxx`` implementations of C++17 polyfills by default when no polyfill library is
+  requested. This option is ``OFF`` by default.
+- The CMake option ``BSONCXX_POLY_USE_IMPLS`` allows selecting ``bsoncxx`` implementations
+  of C++17 polyfills instead of external libraries or the C++ standard library. This
+  option is ``OFF`` by default.
+- The ``bsoncxx::v_noabi`` and ``mongocxx::v_noabi`` namespaces are no longer declared inline.
+  Root namespace declarations are still supported and expected to remain equivalent to
+  their prior definitions. For example, ``bsoncxx::document`` is still equivalent to
+  ``bsoncxx::v_noabi::document``, and ``mongocxx::client`` is still equivalent to
+  ``mongocxx::v_noabi::client``.
+- By default, when library filenames are compiled with MSVC (as detected by CMake's ``MSVC``
+  variable), they are embedded with an ABI tag string. These strings are similar to
+  the following:
+  
+  .. code-block:: bash
+
+     bsoncxx-v_noabi-rhs-x64-v142-md.lib
+
+  To disable this behavior, set ``ENABLE_ABI_TAG_IN_LIBRARY_FILENAMES=OFF`` as part of
+  your CMake configuration.
+
+  The ABI tag string can also be embedded in pkg-config metadata filenames, similar to the
+  following:
+    
+  .. code-block:: bash
+      
+     libbsoncxx-v_noabi-rhs-x64-v142-md.pc
+       
+  To enable this behavior, set ``ENABLE_ABI_TAG_IN_LIBRARY_FILENAMES=ON`` and
+  ``ENABLE_ABI_TAG_IN_PKGCONFIG_FILENAMES=ON`` as part of your CMake configuration.
+
+To learn more about this release, see the
+`v3.10 Release Notes <https://github.com/mongodb/mongo-cxx-driver/releases/tag/r3.10.0>`__
+on GitHub.
+
+.. _version-3.9:
+
+What's New in 3.9
+-----------------
+
+.. warning:: Breaking Changes
```

Comment:
```suggestion
.. warning:: Build System Breaking Changes
```
These too are breaking changes pertaining only to the build system (not API or ABI).